src/coreclr/src/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -61,11 +62,15 @@ private static string MarshalToString(IntPtr arg, string argName)
                 string typeName = MarshalToString(typeNameNative, nameof(typeNameNative));
                 string methodName = MarshalToString(methodNameNative, nameof(methodNameNative));
 
-                string delegateType;
+                string? delegateType;
                 if (delegateTypeNative == IntPtr.Zero)
                 {
                     delegateType = typeof(ComponentEntryPoint).AssemblyQualifiedName!;
                 }
+                else if (delegateTypeNative == (IntPtr)(-1))
```

Comment:
I do not think you can treat this sentinel the same way as NULL name. It will change how to API behaves for NULL name.

Reply:
I think it would be cleaner to pass in a known string value. Since the value is normally a fully qualified type name, we can easily create a dummy type (which may even exist, or maybe it should not) which acts as the "sentinel". Performance wise it should not hurt that much (marshalling one string and a simple string compare).

Reply:
>I do not think you can treat this sentinel the same way as NULL name. It will change how to API behaves for NULL name.

I don't understand this statement. NULL is handled first, then we check if it is the sentinel.

>I think it would be cleaner to pass in a known string value. Since the value is normally a fully qualified type name, we can easily create a dummy type (which may even exist, or maybe it should not) which acts as the "sentinel". Performance wise it should not hurt that much (marshalling one string and a simple string compare).

I'm not against that, but I don't see the reason. We need to some up with a string value that is never a type, not terribly hard, then we need to create some constant in the header, which is more annoying than hard. The -1 value is descriptive and easy to identify in a DMP as opposed to a string which means it needs to be examined rather than observed in a register or local. There is no real user value here between the two, only DMP analysis. Is there a specific reason there is a preference for a string from a UX perspective?

Reply:
> I don't understand this statement. NULL is handled first, then we check if it is the sentinel.

Ok, you are right. I got confused by `delegateTypeNative` vs. `delegateType`

Reply:
OK - -1 will work... I personally like the string sentinel a little bit better, but I'm fine with -1 (I'm not a fan of magical pointer values in general - probably ever since I had to deal with the least important function pointer bit on ARM).